Funcky.SourceGenerator/OrNoneFromTryPatternGenerator.cs

```diff
@@ -29,34 +28,25 @@ private static SourceProductionContext CreateSourceByClass(SourceProductionConte
     {
         var syntaxTree = OrNoneFromTryPatternPartial.GetSyntaxTree(methodByClass.First().NamespaceName, methodByClass.First().ClassName, methodByClass.SelectMany(m => m.Methods));
 
-        context.AddSource($"{Path.GetFileName(methodByClass.Key)}.g.cs", string.Join(Environment.NewLine, GeneratedFileHeadersSource) + Environment.NewLine + syntaxTree.NormalizeWhitespace().ToFullString());
+        context.AddSource($"{Path.GetFileName(methodByClass.Key)}.g.cs", string.Join("\n", GeneratedFileHeadersSource) + "\n" + syntaxTree.NormalizeWhitespace().ToFullString());
```

Comment:
New analyzer rule says: 
```
RS1035: The symbol 'Environment' is banned for use by analyzers: Analyzers should not read their settings directly from environment variables
```

Reply:
Yeah I ran into that one before: https://github.com/dotnet/roslyn-analyzers/issues/6467#issuecomment-1689875009

The reason is, the build is not deterministic if you change the new line. (That is the official argument.)